iis-common/src/test/java/eu/dnetlib/iis/common/java/jsonworkflownodes/SelectiveTestingConsumer.java

```diff
@@ -0,0 +1,144 @@
+        Preconditions.checkArgument(StringUtils.isNotBlank(propertiesPathsCSV), 
+                "no '%s' property value provided, field requirements were not specified!", PROPERTIES_CSV);
+
+        String[] recordsExpectationPropertiesLocations = StringUtils.split(propertiesPathsCSV, ',');
+        FileSystem fs = FileSystem.get(configuration);
+
+        if (!fs.exists(inputRecordsPath)) {
+            throw new RuntimeException(inputRecordsPath + " hdfs location does not exist!");
+        }
+        
+        try (CloseableIterator<SpecificRecord> recordsIterator = DataStore.getReader(new FileSystemPath(fs, inputRecordsPath))) {
+            
+            int recordsCount = 0;
+            
+            while (recordsIterator.hasNext()) {
+                
+                SpecificRecord record = recordsIterator.next();
+                recordsCount++;
+
+                if (recordsCount > recordsExpectationPropertiesLocations.length) {
+                    throw new RuntimeException("got more records than expected: " + "unable to verify record no " + recordsCount
+                            + ", no field specification provided! Record contents: " + JsonUtils.toPrettyJSON(record.toString()));
+                } else {
+                    validateRecord(record, readProperties(recordsExpectationPropertiesLocations[recordsCount - 1]));    
+                }
+            }
+            
+            if (recordsCount < recordsExpectationPropertiesLocations.length) {
+                throw new RuntimeException(
+                        "records count mismatch: " + "got: " + recordsCount + " expected: " + recordsExpectationPropertiesLocations.length);
+            }
+        }
+    }
+    
+    //------------------------ PRIVATE ---------------------------------
+    
+    /**
+     * Reads properties from given location.
+     */
+    private static Properties readProperties(String location) throws IOException {
+        log.info("fields expectations location: " + location);
+        Properties properties = new OrderedProperties();
+        properties.load(TestingConsumer.class.getResourceAsStream(location.trim()));
+        return properties;
+    }
+    
+    /**
+     * Validates record fields against specified expectations. RuntimeException is thrown when invalid.
+     * 
+     * @param record avro record to be validated
+     * @param recordFieldExpectations set of field expectations defined as properties where key is field location and value is expected value
+     */
+    private static void validateRecord(SpecificRecord record, Properties recordFieldExpectations) throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Iterator<Entry<Object,Object>> expectationPropertiesIterator = recordFieldExpectations.entrySet().iterator();
+        while (expectationPropertiesIterator.hasNext()) {
+            Entry<Object,Object> fieldExpectation = expectationPropertiesIterator.next();
```

Comment:
Would it be clearer to have:
    for (Entry<String, String> fieldExpectation : (Entry<String, String>)recordFieldExpectations.entrySet()) {
    }